test/NuGet.Core.Tests/NuGet.PackageManagement.Test/BuildIntegration/DependencyGraphProjectCacheUtilityTests.cs

```diff
@@ -0,0 +1,567 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Frameworks;
+using NuGet.PackageManagement;
+using NuGet.PackageManagement.Test;
+using NuGet.ProjectManagement.Projects;
+using NuGet.ProjectModel;
+using NuGet.Test.Utility;
+using Test.Utility;
+using Xunit;
+
+namespace NuGet.Test
+{
+    public class DependencyGraphProjectCacheUtilityTests
+    {
+        [Fact]
+        public void DependencyGraphProjectCacheUtility_GetExternalClosure_Empty()
```

Comment:
Can you also include tests for msbuildprojectsystem and mix of buildintegrated with msbuild project system? I only see buildintegratedprojectsystem in all these tests.